x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -983,6 +984,38 @@ public LogicalPlan apply(LogicalPlan p) {
         }
     }
 
+    // This is a workaround for the https://github.com/elastic/elasticsearch/issues/45251 issue
+    // Should be removed as soon as the above issue is fixed
+    // NOTE: this rule should always be applied AFTER the ReplaceAggsWithStats rule
```

Comment:
I find this comment confusing.
This PR is a fix for https://github.com/elastic/elasticsearch/issues/45251 itself due to an issue in Elasticsearch not SQL.
Once the issue in Elasticsearch is fixed the workaround in QL won't be needed. Pointing to the SQL issue itself asking for the workaround is incorrect.

Reply:
I intended to keep #45251 open after this PR and only close it after the `sum` aggregations can differentiate between `SUM(all zeroes)` and `SUM(all nulls)`. Reworded the comment to make this more obvious.